src/raftstore/store/fsm/peer.rs

```diff
@@ -2341,6 +2341,18 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
             self.ctx.raft_metrics.invalid_proposal.mismatch_peer_id += 1;
             return Err(e);
         }
+        if !self.fsm.peer.is_initialized() {
```

Comment:
should check it in `read.rs` too?

Reply:
Also, please add comments

Reply:
This check should apply to all proposals, not only read.

Reply:
If the proposal is processed by the local_reader, it wouldn't call here.

Reply:
It also applies to write cmd if we enable follower to redirect proposals in the future. And the key problem is `LocalReader` does not have any information about peer.😥️

Reply:
How local reader handles follower read?

Reply:
follower read will potentially be rejected by two conditions in LocalReader:
1. `ReadDelegate.applied_index_term != ReadDelegate.term`. `applied_index_term` is  only updated after the leader applies a proposal, `term` is only updated when peer transfers to leader
2. lease is expired

Reply:
How can follower have lease?

Reply:
> How can follower have lease?

Then follower read will always be rejected and redirected to raftstore,